src/routes/partRoutes.js

```diff
@@ -0,0 +1,59 @@
+import { Router } from "express";
+import { 
+  createPart, getPartById, searchPartsByName, getCarsByPartId 
+} from '../data/parts.js';
+import { auth } from '../middleware/auth.js';
+import { error } from '../middleware/error.js';
+
+const router = express.Router();
+
+router.get("/", async (req, res, next) => {
+  try {
+    const experts = await expertDb.getAllExperts();
+    res.status(200).send(experts);
+  } catch (e) {
+    next(e);
+  }
+})
+  .post('/', auth, async (req, res, next) => {
+    try {
+        const { name, price, manufacturer, sellerId, carIds } = req.body;
+        const newPart = await createPart(name, price, manufacturer, sellerId, carIds);
+        res.status(201).json(newPart);
+    } catch (e) {
+        next(e);
+    }
+});
+
+
+router.get('/:partId', auth, async (req, res, next) => {
+    try {
+        const part = await getPartById(req.params.partId);
+        res.status(200).json(part);
+    } catch (e) {
+        next(e);
+    }
+});
+
+router.get('/search', auth, async (req, res, next) => {
+    try {
+        const { query } = req.query;
+        const parts = await searchPartsByName(query);
+        res.render('partSearch', { results: parts });
```

Comment:
add user: req.user to the data for all render
res.render('partSearch', { results: parts, user: req.user });